### ISSUE TITLE
New stylelint `function-no-unknown` rule not working well with SCSS

### ISSUE THREAD

**Issue by dcwarwick** [stylelint-scss/stylelint-scss#589]
A new rule `function-no-unknown` was recently added to `stylelint`, merged by @jeddy3 for stylelint/stylelint#5853. This rule considers functions defined in the CSS Specifications to be 'known', and ignores double-dashed custom functions, _e.g._ `--custom-function()`. This rule seems good and useful, and the recommendation to use double-dashed names so as to avoid clashes with current or future standard CSS names would seem applicable to SASS/SCSS functions. However, I have encountered some problems when applying it in conjunction with `stylelint-config-standard-scss` and linting SCSS files. I don't know whether the issue is one for `stylelint-config-standard-scss` to address or whether it needs addressing in base `stylelint`, or a combination of both, so I have opened this issue here to track it.

### Problem 1

Standard functions defined by SASS are not 'known' by this rule, even if `stylelint-config-standard-scss` is being used. This causes many errors to be produced such as:
- Unexpected unknown function "if"
- Unexpected unknown function "map-get"
- _etc_

This is easily worked around by adding these built-in function names to the rule using `ignoreFunctions`, but that is a manual and error-prone process. SASS built-in functions should be automatically included as 'known' when SASS/SCSS files are being linted.

### Problem 2

The rule does not play well with interpolation, as it doesn't recognise the syntax of `#{...}`. For example, a line such as
```scss
      transform: translateX(calc(-1 * #{map-get($side-panel-sizes, md)}));
```
produces an error such as
- Unexpected unknown function "#{map-get"

This error is poorly formed, and note that this also means that even if the function which appears at the start of the `#{...}` interpolation is a 'known' function or is double-dashed it will still be reported as an error. For example:
- Unexpected unknown function "#{--custom-function".

Again, this is easily worked around by including the names of affected functions including the wrongly-included `#{` into `ignoreFunctions`, but this is rather fiddly and far from ideal. The rule should understand the parsed syntax of SASS interpolation and correctly identify function names within this.

**Comment by jeddy3** [comment]
Thanks for the issue.

>  I don't know whether the issue is one for stylelint-config-standard-scss to address or whether it needs addressing in base stylelint, or a combination of both,

It'll be a combination of both.

> Standard functions defined by SASS are not 'known' by this rule, even if stylelint-config-standard-scss is being used

I've transferred your issue to the [stylelint-scss](https://github.com/stylelint-scss/stylelint-scss/) issue tracker. A new rule can be added here to replace Stylelint's built-in rule, and then used in the `stylelint-config-standard-scss` shared config. I'm sure the maintainers of this plugin would be grateful for a pull request if you've time.

> This is easily worked around by adding these built-in function names to the rule using `ignoreFunctions`,

This new rule can use the `ignoreFunction` option and the [`checkAgainstRule`](https://stylelint.io/developer-guide/plugins#stylelintutilscheckagainstrule) util, much like the [`scss/at-rule-no-unknown` rule does](https://github.com/stylelint-scss/stylelint-scss/blob/master/src/rules/at-rule-no-unknown/index.js).

> The rule does not play well with interpolation, as it doesn't recognise the syntax of `#{...}.`

Please can you [report a bug in the stylelint issue tracker](https://github.com/stylelint/stylelint/issues/new?assignees=&labels=&template=REPORT_A_BUG.yml) as the built-in rule should ignore any functions containing interpolation as it's non-standard.

**Comment by ybiquitous** [comment]
Duplicate of #581?

**Comment by nthornton2010** [comment]
I'm not sure why this was closed. I just installed the latest package and this is still happening.
